### PR TITLE
Use auto-generate metadata in wf2 mappings and improve wf3

### PR DESCRIPTION
**Pull request by mtuchi**
**Description** 
- Update wf2 to use the auto generated metadata file 
-  Update wf3 to fetch use a list of form from `orm-form-metadata` sheet
- Deleted `get-options-map` and move the logic to `2-mappings.js` 
- Update spec.yam file to use file path and remove get-options-map step

**Issue**
- #56

### CODE REVIEW

**Comment by aleksa-krolls** [comment]
hey @josephjclark can we please get your feedback here? I'd like to merge this for testing later this afternoon

**Comment by aleksa-krolls** [comment]
hey @josephjclark can we please get your feedback here? I'd like to merge this for testing later this afternoon

File: workflows/wf2/2-upsert-teis.js [hunk 1]
```diff
@@ -64,73 +86,13 @@ const buildPatientsUpsert = (state, patient, isNewPatient) => {
...
-            a => a.attributeType.uuid === 'e363161a-9d5c-4331-8463-238938f018ed'
-          )?.value, //input.attributeType = "e363161a-9d5c-4331-8463-238938f018ed"
-        },
+        ...statusAttrMaps,
       ],
     },
   };
 
-  console.log('mapped dhis2 payloads:: ', JSON.stringify(payload, null, 2));
+  // TODO: AK do we need this log👇🏾?
```
Comment (josephjclark) [inline workflows/wf2/2-upsert-teis.js:94]: Paging @aleksa-krolls

File: workflows/wf2/2-upsert-teis.js [hunk 2]
```diff
@@ -143,7 +105,7 @@ const buildPatientsUpsert = (state, patient, isNewPatient) => {
 const delay = ms => new Promise(resolve => setTimeout(resolve, ms));
 
 each(
-  '$.patients[*]',
+  $ => $.patients.slice(0, 1),
```
Comment (josephjclark) [inline workflows/wf2/2-upsert-teis.js:108]: Hello, what's this?

File: workflows/wf3/1-fetch-metadata.js [hunk 3]
```diff
@@ -38,10 +24,31 @@ fn(state => {
   return state;
 });
 
+get(
+  `${$.workbookBase}/worksheets('omrs-form-metadata')/usedRange`,
+  {},
+  state => {
+    const [headers, ...rows] = state.data.values.slice(0);
```
Comment (josephjclark) [inline workflows/wf3/1-fetch-metadata.js:31]: Why the slice here?

File: workflows/wf3/1-fetch-metadata.js [hunk 4]
```diff
@@ -38,10 +24,31 @@ fn(state => {
...
+    state.formMetadata = rows
+      .map(row =>
+        row.reduce((obj, value, index) => {
+          if (value != null && value !== '') {
```
Comment (josephjclark) [inline workflows/wf3/1-fetch-metadata.js:35]: You should be able to just do `if(value)` here, no?

File: workflows/wf2/2-upsert-teis.js (hunk 1 above)
Comment (aleksa-krolls) [reply workflows/wf2/2-upsert-teis.js:94]: @mtuchi leave the log in for now - just for testing

File: workflows/wf3/1-fetch-metadata.js (hunk 3 above)
Comment (mtuchi) [reply workflows/wf3/1-fetch-metadata.js:31]: To separate the header row from other rows with content

File: workflows/wf3/1-fetch-metadata.js (hunk 4 above)
Comment (mtuchi) [reply workflows/wf3/1-fetch-metadata.js:35]: I will have a look

File: workflows/wf3/1-fetch-metadata.js (hunk 3 above)
Comment (josephjclark) [reply workflows/wf3/1-fetch-metadata.js:31]: But why not ` const [headers, ...rows] = state.data.values` ?

File: workflows/wf2/2-upsert-teis.js (hunk 2 above)
Comment (mtuchi) [reply workflows/wf2/2-upsert-teis.js]: Resolved

File: workflows/wf3/1-fetch-metadata.js (hunk 3 above)
Comment (mtuchi) [reply workflows/wf3/1-fetch-metadata.js]: removed `.slice(0)`

**Comment by mtuchi** [comment]
Hiya @josephjclark worked on your feedback, Please have another look

**Comment by josephjclark** [comment]
wf2 and wf3 are both modified in this PR - the title should really be changed to help with later audits

File: workflows/wf2/2-upsert-teis.js (hunk 1 above)
Comment (josephjclark) [reply workflows/wf2/2-upsert-teis.js:90]: Sorry @mtuchi can you please remove the TODO from here?

**Review by josephjclark — APPROVED** [review]

**Comment by mtuchi** [comment]
Updated the PR title, merging this now

**Pull request merged**